pkg/apis/datadoghq/v1alpha1/datadogagent_validation.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	utilserrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// IsValidDatadogAgent use to check if a DatadogAgentSpec is valid
+func IsValidDatadogAgent(spec *DatadogAgentSpec) error {
```

Comment:
should this be renamed `IsValidCustomConfig` ? as we're only validating the `CustomConfig` fields of each component - I liked the idea of having a validation mechanism of the spec that could be extended for all fields but maybe it's not in the scope of this PR

Reply:
Maybe we can keep the function name and add a TODO to validate more fields

Reply:
we may have other fields to validate late.